core/vulkan/vk_virtual_swapchain/cc/virtual_swapchain.cpp

```diff
@@ -294,6 +305,64 @@ void VirtualSwapchain::Destroy(const VkAllocationCallbacks* pAllocator) {
   functions_->vkDestroyCommandPool(device_, command_pool_, pAllocator);
 }
 
+void VirtualSwapchain::DumpImageToFile(uint8_t* image_data, size_t size) {
+  std::unique_ptr<char[]> image_data_owned(new char[size]());
+  memcpy(image_data_owned.get(), image_data, size);
+
+  auto now = std::chrono::system_clock::now().time_since_epoch().count();
+  auto image_path = image_dump_dir_ + "/image_" +
+                    std::to_string(dumped_frame_count_++) + "_ts_" +
+                    std::to_string(now) + ".ppm";
+
+  auto ppm_writer = [](std::unique_ptr<char[]> image_data, size_t size,
+                       std::string file_name, uint32_t width, uint32_t height,
+                       VkFormat image_format) {
+    std::ofstream out(file_name, std::ios::out | std::ios::binary);
+    if (!out) {
+      std::cout << "Create file " << file_name << " failed." << std::endl;
```

Comment:
cerr?

Reply:
More generally, std::cout is irrelevant on Android... 
There does not seem to be a single print/log in the virtual swapchain code, maybe for the sake of portability, so maybe it's OK to not output anything here?